velox/exec/AggregateWindow.cpp

```diff
@@ -341,7 +347,26 @@ class AggregateWindowFunction : public exec::WindowFunction {
     });
 
     // Set null values for empty (non valid) frames in the output block.
-    setNullEmptyFramesResults(validRows, resultOffset, result);
+    setEmptyFramesResult(validRows, resultOffset, result, emptyResult_);
+  }
+
+  // Precompute and save the aggregate output for empty input in emptyResult_.
+  // This value is returned for rows with empty frames.
+  void computeInitAggregateValue(const TypePtr& resultType) {
+    aggregate_->clear();
+    aggregate_->initializeNewGroups(
+        &rawSingleGroupRow_, std::vector<vector_size_t>{0});
+
+    // addSingleGroupRawInput is called on an empty group to catch the expected
```

Comment:
Could you use teh same rawSingleGroupRow_ here instead of emptyGroup ? It seems like the aggregate logic (maybe unintentionally) prefers addSingle(Group)RawInput() called atleast once before extract result values from it.

Reply:
Thanks for the suggestion, I tried this before and it results in the average aggregate returning incorrect results for some test cases because of the new non-null group.